src/gwt/acesupport/acemode/markdown_highlight_rules.js

```diff
@@ -165,8 +165,36 @@ var MarkdownHighlightRules = function() {
             emphasisStars,
             emphasisUnderscore
         ],
-       
+
+        "criticmarkup" : [{ // CriticMarkup syntax (http://criticmarkup.com/spec.php)
+            token : "criticmarkup_insertion",
+            regex : "\\{\\+\\+",
+            next  : "cm-insertion"
```

Comment:
I think this should be `push` rather than `next`. Ace can maintain the set of states as a 'stack', so you can write in one rule:

```
push : "my-state"
```

And then a rule in `"my-state"` can use:

```
next: "pop"
```

to return to the original state before entering `"my-state"`.